src/tests/mem/mmap.c

```diff
@@ -36,7 +36,7 @@ static void *mmap_test_child(void *arg) {
 	int allocated = 0;
 	while (1) {
 		void *ptr = mmap(NULL, PAGE_SIZE(), PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANONYMOUS, -1, 0);
-		if (!ptr)
+		if (ptr != MAP_FAILED)
```

Comment:
It was `if (!ptr)`. So I think it should be `ptr == MAP_FAILED`, not `ptr != MAP_FAILED`.
The test logic, as I can see, suppose that we break only when all pages (per task) were already mapped.